app/lib/actions/pulp3/orchestration/repository/upload_content.rb

```diff
@@ -7,8 +7,14 @@ class UploadContent < Pulp3::Abstract
           def plan(repository, smart_proxy, file, unit_type_id)
             sequence do
               content_backend_service = SmartProxy.pulp_primary.content_service(unit_type_id)
-              content_list = content_backend_service.content_api.list("sha256": Digest::SHA256.hexdigest(File.read(file[:path])))
-              content_href = content_list&.results&.first&.pulp_href
+              if repository.generic?
+                # pulp python doesn't support filter by sha256 (i.e. list method), so need to get the result differently
+                content_list = content_backend_service.content_api(repository.repository_type, unit_type_id).list.results.find { |result| result.sha256 == Digest::SHA256.hexdigest(File.read(file[:path])) }
```

Comment:
I don't think this will work (or at least won't work if there is more than a page worth of python modules).

It looks like this was now added:  https://github.com/pulp/pulp_python/commit/aeb205db4a9cf02a6c65674894832cb5bad5295e

maybe we could just go ahead and patch locally for testing and ask them for a release?

Reply:
There's an optional parameter for specifying the amount of results. I can't find what happens if that number isn't specified, but it isn't necessarily all of them. But yeah I can change it to depend on that change.